src/utils/serve-functions.js

```diff
@@ -484,9 +498,9 @@ const setupFunctionsBuilder = async ({ config, errorExit, functionsDirectory, lo
 
   const functionWatcher = chokidar.watch(functionBuilder.src)
   functionWatcher.on('ready', () => {
-    functionWatcher.on('add', debouncedBuild)
-    functionWatcher.on('change', debouncedBuild)
-    functionWatcher.on('unlink', debouncedBuild)
+    functionWatcher.on('add', (path) => debouncedBuild(path, 'add'))
```

Comment:
The build function now receives the path of the updated file _and_ the type of event.